packages/node_modules/@node-red/nodes/core/function/90-exec.js

```diff
@@ -61,12 +63,16 @@ module.exports = function(RED) {
             }
             else {
                 var child;
+                // make the extra args into an array
+                // then prepend with the msg.payload
+                var arg = node.cmd;
+                if (node.addpay) {
+                    if (node.addpayValueType === "msg" && msg.hasOwnProperty(node.addpayValue)) {
```

Comment:
This will fail if the user specifies a property below the top level. For example `msg.payload[0].something`.

You use use `RED.util.getMessageProperty` to get the property value and check it's not undefined.

Reply:
I tried `msg.payload[0].something`. It failed. In the commit, https://github.com/node-red/node-red/pull/2908/commits/89485971fa005a963d5f65c7cd7c87c1ab24c977, I used `RED.util.getMessageProperty()` to check value.